frontend/src/pages/offers/ViewOfferPage.tsx

```diff
@@ -29,6 +36,38 @@ const ViewOfferPage: FunctionComponent = () => {
     variables: { id: sendingGroupId! },
   })
 
+  const [updateOffer] = useUpdateOfferMutation()
+
+  const updateOfferStatus = (newStatus: OfferStatus) => {
+    if (!offer) {
+      return
+    }
+
+    const updatedOffer: OfferUpdateInput = {
+      id: offer.offer.id,
+      status: newStatus,
+      contact: _omit(offer.offer.contact, ['__typename']),
```

Comment:
suggestion (non-blocking): consider extracting a shared function for this, using it here and [here](https://github.com/distributeaid/shipment-tracker/blob/dae4afa21f130cdf0296bab8e83a873c46777220/frontend/src/pages/groups/GroupForm.tsx#L48)